monai/transforms/croppad/batch.py

```diff
@@ -101,17 +100,22 @@ def __call__(self, batch: Any):
 
                 # If we have a dictionary of data, append to list
                 if is_list_of_dicts:
-                    self.push_transform(batch[idx], key_or_idx, orig_size=orig_size)
+                    self.push_transform(
+                        batch[idx],
+                        key_or_idx,
+                        orig_size=orig_size,
+                        extra_info=self.pop_transform(batch[idx], key_or_idx, check=False),
```

Comment:
Could you please help add some comments here why `pop & push` to help open-source readers understand the logic and in case we all forget in the future...

Thanks.